### PR TITLE
feat: environment/tenant security principal and role management commands

### PR DESCRIPTION
**Pull request by TomProkop**
## Summary

Adds `txc environment user/service-principal/team/role` and `txc tenant role/service-principal/user/group` commands for managing Dataverse security principals (systemusers, application users/service principals, teams) and their security role assignments, at both environment (Dataverse) and tenant (Power Platform admin) scope.

By design this CLI does **not** mutate Entra ID — no create/delete of users, groups, or app registrations. It assumes those identities already exist and only manages the Dataverse/Power Platform side (provisioning systemusers/application users, team membership, role assignment).

## Command groups

**`txc environment ...`**
- `user list/get/add/update/role list/role add/role remove/self-elevate`
- `service-principal list/get/create/update/delete/role list/role add/role remove`
- `team list/get/create/delete/member list/member add/member remove/role list/role add/role remove`
- `role list/get`

**`txc tenant ...`**
- `role list/get`
- `user role list/add/remove`
- `service-principal role list/add/remove`
- `group role list/add/remove` (group identified by Entra **object id only**, not display name — see below)

## Notable decisions

- **Named `service-principal`, not `app`.** `app` is reserved for a future `txc environment app` command group covering actual Power Apps (canvas/model-driven apps). The underlying Dataverse service/record types still say "Application User" since that's the accurate Dataverse term for the `systemuser` record; only the CLI command surface uses `service-principal`.
- **`tenant group` never calls Microsoft Graph.** Graph group search requires `Group.Read.All`, which isn't pre-consented for this CLI's Entra app registration in most tenants (confirmed via a real 403 in live testing). Following `pac admin assign-group`'s approach, `--group` is always a raw object id; `tenant group list/get` were removed since they have no non-Graph equivalent.
- Microsoft Graph rejects `$filter` entirely (400) if any clause compares a GUID-typed property to a non-GUID literal — filters are now built to only include `id`/`appId` clauses when the input parses as a GUID.
- `environment team role add/remove` only supports owner/AAD-security-group/AAD-office-group teams, not access teams (Dataverse limitation, now reflected in help text).

## Out of scope

Entra ID mutation (create/delete of users, groups, app registrations) — this CLI assumes those identities already exist.

## Testing

698 tests passing (unit + regression coverage for idempotency, role resolution, and OData filter edge cases).

### CODE REVIEW

File: src/TALXIS.CLI.Features.Tenant/User/UserCommandSupport.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        var escaped = TenantPrincipalCommandSupport.EscapeODataString(filter.Trim());
+        return $"startswith(userPrincipalName,'{escaped}') or startswith(displayName,'{escaped}')";
+    }
+
+    private static string BuildGetFilter(string user)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(user);
+        var escaped = TenantPrincipalCommandSupport.EscapeODataString(user.Trim());
+        return $"id eq '{escaped}' or userPrincipalName eq '{escaped}'";
+    }
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Features.Tenant/User/UserCommandSupport.cs:158]: Fixed in 0d83122 — this call site now guards the id/appId clauses to only appear when the input parses as a GUID, mirroring TenantRoleResolver's pattern. Added regression tests covering both GUID and non-GUID inputs, and live-verified against production that the fixed commands no longer 400.

File: src/TALXIS.CLI.Features.Tenant/App/TenantAppCommandSupport.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            return null;
+
+        return $"startswith(displayName,'{EscapeODataString(filter.Trim())}')";
+    }
+
+    private static string BuildExactAppFilter(string app)
+    {
+        var escaped = EscapeODataString(app.Trim());
+        return $"appId eq '{escaped}' or id eq '{escaped}' or displayName eq '{escaped}'";
+    }
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Features.Tenant/App/TenantAppCommandSupport.cs]: Fixed in 0d83122 — this call site now guards the id/appId clauses to only appear when the input parses as a GUID, mirroring TenantRoleResolver's pattern. Added regression tests covering both GUID and non-GUID inputs, and live-verified against production that the fixed commands no longer 400.

File: tests/TALXIS.CLI.Tests/Tenant/User/UserCliCommandTests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        Assert.Equal("alice@contoso.com", users[0].GetProperty("userPrincipalName").GetString());
+    }
+
+    [Fact]
+    public async Task RunAsync_Get_MissingUser_ReturnsValidationError()
+    {
+        using var host = new TenantPrincipalCommandTestHost(new Queue<Func<HttpRequestMessage, HttpResponseMessage>>([
+            request =>
+            {
+                Assert.Contains("$filter=id eq 'missing@contoso.com' or userPrincipalName eq 'missing@contoso.com'", Uri.UnescapeDataString(request.RequestUri!.Query));
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply tests/TALXIS.CLI.Tests/Tenant/User/UserCliCommandTests.cs]: Fixed in 0d83122 — this call site now guards the id/appId clauses to only appear when the input parses as a GUID, mirroring TenantRoleResolver's pattern. Added regression tests covering both GUID and non-GUID inputs, and live-verified against production that the fixed commands no longer 400.

File: src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+/// environment. Mirrors <c>TenantPrincipalCommandSupport</c> on the tenant
+/// side.
+/// </summary>
+internal static class EnvironmentPrincipalCommandSupport
+{
+    /// <summary>
+    /// Resolves the mutually-exclusive <c>--enabled</c>/<c>--disabled</c>/<c>--all</c>
+    /// list filter options shared by <c>environment user list</c> and
+    /// <c>environment app list</c>.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs:19]: Fixed in c41dfa0 — updated the doc comment to reference `service-principal` instead of the stale `app` wording.

File: src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                ? DataverseSecurityPrincipalStateFilter.All
+                : DataverseSecurityPrincipalStateFilter.Enabled;
+        return true;
+    }
+
+    /// <summary>
+    /// Parses a comma-separated <c>--role</c> option value into a
+    /// deduplicated list of role names/GUIDs, shared by the environment user
+    /// and app "create with roles" commands.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs:47]: Fixed in c41dfa0 — updated the doc comment to reference `service-principal` instead of the stale `app` wording.

File: src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    /// <summary>
+    /// Shared exit-code mapping for the validation-style exceptions raised by
+    /// Dataverse security-principal resolution/mutation (ambiguous friendly-name
+    /// match, invalid argument, invalid operation). Callers supply a
+    /// <paramref name="logAmbiguousMatch"/> delegate so each command group can
+    /// keep its own candidate-listing format (e.g. <c>environment user</c> logs a
+    /// bulleted list, <c>environment app</c> logs a single "Candidate:" line per
+    /// match) while sharing the exception-type dispatch and exit-code contract.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Features.Environment/EnvironmentPrincipalCommandSupport.cs:101]: Fixed in c41dfa0 — updated the doc comment to reference `service-principal` instead of the stale `app` wording.

File: src/TALXIS.CLI.Platform.PowerPlatform.Control/Strategies/BapAdminApplicationRoleStrategy.cs [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+/// <summary>
+/// Implements the synthetic <c>admin-application</c> tenant role by registering/deregistering
+/// the application with the Power Platform Admin (BAP) API's <c>adminApplications</c> endpoint,
+/// the mechanism Microsoft documents at
+/// <see href="https://learn.microsoft.com/en-us/power-platform/admin/powerplatform-api-create-service-principal">
+/// Create a service principal to create and manage environments and other resources for Power
+/// Platform</see>. Unlike every other tenant role, this is not a Power Platform RBAC role
+/// assignment - <see cref="PowerPlatformTenantRoleAssignment.IsSynthetic"/> is always
+/// <see langword="true"/> for assignments produced by this strategy, so callers can tell it
+/// apart from real RBAC assignments in <c>tenant app role list</c> output.
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Platform.PowerPlatform.Control/Strategies/BapAdminApplicationRoleStrategy.cs]: Fixed in c41dfa0 — updated the doc comment to reference `service-principal` instead of the stale `app` wording.

File: src/TALXIS.CLI.Platform.PowerPlatform.Control/Graph/GraphODataFilterSupport.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
+namespace TALXIS.CLI.Platform.PowerPlatform.Control.Graph;
+
+/// <summary>
+/// Shared Microsoft Graph OData <c>$filter</c> construction helpers used by
+/// every <c>txc tenant</c> command that resolves a user or application by a
+/// caller-supplied GUID-or-friendly-name identifier (e.g. <c>--user</c>,
+/// <c>--app</c>).
+/// </summary>
```
[the remark this answers is not in the document]
Comment (TomProkop) [reply src/TALXIS.CLI.Platform.PowerPlatform.Control/Graph/GraphODataFilterSupport.cs:8]: Fixed in c41dfa0 — updated the doc comment to reference `service-principal` instead of the stale `app` wording.
